### PR TITLE
[IMP] fleet : design and demo data improvement

### PR DESCRIPTION
**Pull request by mepe-odoo**
With this commit:
	- a user can choose the unit for his engine (horse/Kw)
	- fix model year display
	- a user can change his vehicle image
	- information linked to a car are hidden for a bike.

task: 3640463

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/fleet/data/fleet_cars_data.xml [hunk 1]
```diff
@@ -1,492 +1,378 @@
...
+        <field name="name">BMX</field>
+    </record>
+    <record id="model_category_vtt" model="fleet.vehicle.model.category">
+        <field name="name">VTT</field>
+    </record>
+    <record id="model_category_city" model="fleet.vehicle.model.category">
+        <field name="name">City</field>
+    </record>
+
+    <record id="model_corolla_ts" model="fleet.vehicle.model">
```
Comment (tivisse) [inline addons/fleet/data/fleet_cars_data.xml:299]: It's a good move to drop all the data vehicle models, but why adding this as new data ? Probably unrelated to end users need + nobody is gonna maintain this, resulting into outdated data in a few versions.

File: addons/fleet/models/fleet_vehicle.py [hunk 2]
```diff
@@ -123,6 +127,12 @@ def _compute_service_activity(self):
             activities_state = set(state for state in vehicle.log_services.mapped('activity_state') if state and state != 'planned')
             vehicle.service_activity = sorted(activities_state)[0] if activities_state else 'none'
 
+    @api.depends('model_id')
+    def _compute_image_1920(self):
+        for vehicle in self:
+            if not vehicle.image_1920:
+                vehicle.image_1920 = vehicle.model_id.image_1920
```
Comment (tivisse) [inline addons/fleet/models/fleet_vehicle.py:134]: Why not a related ? Is it really important to keep track of the vehicle image at the time ?

**Review by tivisse — COMMENTED** [review]
Small questions, otherwise looks good. 

Thanks.

File: addons/fleet/views/fleet_vehicle_model_views.xml [hunk 3]
```diff
@@ -37,16 +44,14 @@
                                 <group string="Model" invisible="vehicle_type != 'car'">
                                     <field name="seats"/>
                                     <field name="doors"/>
-                                    <field name="color"/>
+                                    <field name="color" invisible="vehicle_type == 'car'"/>
```
Comment (jugj-odoo) [inline addons/fleet/views/fleet_vehicle_model_views.xml:47]: According to the spec color should not be in the view no ?

File: addons/fleet/views/fleet_vehicle_views.xml [hunk 4]
```diff
@@ -117,7 +117,7 @@
                             <group>
                                 <group col="1" name="fiscality_group" string="Fiscality">
                                     <group col="1" name="fiscality_first_group">
-                                        <field name="horsepower_tax" widget="monetary"/>
+                                        <field name="horsepower_tax" widget="monetary" invisible="power_unit != 'horsepower'"/>
```
Comment (jugj-odoo) [inline addons/fleet/views/fleet_vehicle_views.xml:120]: Fiscal horsepower is not related to the unit of measure but to the power of the car in general so it should always be displayed

**Review by jugj-odoo — CHANGES_REQUESTED** [review]
Hello, well done, little comments :sunglasses:

File: addons/fleet/views/fleet_vehicle_model_views.xml (hunk 3 above)
Comment (mepe-odoo) [reply addons/fleet/views/fleet_vehicle_model_views.xml:47]: Geoffrey's review : Color on model should be hidden if vehicle type = car ; I think I can remove the field :thinking:

File: addons/fleet/models/fleet_vehicle_model.py [hunk 5]
```diff
@@ -27,7 +27,7 @@ class FleetVehicleModel(models.Model):
     brand_id = fields.Many2one('fleet.vehicle.model.brand', 'Manufacturer', required=True, tracking=True)
     category_id = fields.Many2one('fleet.vehicle.model.category', 'Category', tracking=True)
     vendors = fields.Many2many('res.partner', 'fleet_vehicle_model_vendors', 'model_id', 'partner_id', string='Vendors')
-    image_128 = fields.Image(related='brand_id.image_128', readonly=True)
+    image_1920 = fields.Image(related='brand_id.image_1920', readonly=True)
```
[the remark this answers is not in the document]
Comment (mepe-odoo) [reply addons/fleet/models/fleet_vehicle_model.py:30]: Geoffrey's spec : 
On the vehicle
Allow to add an image of the vehicle

So I think, He wants the brand's image for the model, and for the vehicle; I want to put the "real" image of the vehicle. So no; I think this field shouldn't be etitable :)

File: addons/fleet/models/fleet_vehicle.py [hunk 6]
```diff
@@ -125,7 +129,7 @@ def _compute_service_activity(self):
             activities_state = set(state for state in vehicle.log_services.mapped('activity_state') if state and state != 'planned')
             vehicle.service_activity = sorted(activities_state)[0] if activities_state else 'none'
 
-    @api.depends('model_id')
+    @api.onchange('model_id')
```
Comment (tivisse) [inline addons/fleet/models/fleet_vehicle.py:132]: Onchanges are only executed in the interface, to revert :)

File: addons/fleet/models/fleet_vehicle_model.py [hunk 7]
```diff
@@ -88,12 +92,20 @@ def _search_vehicle_count(self, operator, value):
 
     def action_model_vehicle(self):
         self.ensure_one()
+        context = {'default_model_id': self.id}
+        if self.vehicle_count == 0:
```
Comment (tivisse) [inline addons/fleet/models/fleet_vehicle_model.py:96]:
```suggestion
        if not self.vehicle_count:
```

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**